soc/intel/intel_adsp/common/include/intel_adsp_hda.h

```diff
@@ -164,14 +164,22 @@ static inline int intel_adsp_hda_set_buffer(uint32_t base,
 	uint32_t aligned_size = buf_size & HDA_BUFFER_SIZE_MASK;
 
 	__ASSERT(aligned_addr == addr, "Buffer must be 128 byte aligned");
-	__ASSERT(aligned_addr >= L2_SRAM_BASE
-		 && aligned_addr < L2_SRAM_BASE + L2_SRAM_SIZE,
-		 "Buffer must be in L2 address space");
 	__ASSERT(aligned_size == buf_size,
 		 "Buffer must be 16 byte aligned in size");
 
-	__ASSERT(aligned_addr + aligned_size < L2_SRAM_BASE + L2_SRAM_SIZE,
-		 "Buffer must end in L2 address space");
+#if defined(CONFIG_KERNEL_VM_SUPPORT)
+#  define _INTEL_ADSP_BASE  CONFIG_KERNEL_VM_BASE
+#  define _INTEL_ADSP_SIZE  CONFIG_KERNEL_VM_SIZE
+#else
+#  define _INTEL_ADSP_BASE  CONFIG_SRAM_BASE_ADDRESS
+#  define _INTEL_ADSP_SIZE  CONFIG_SRAM_SIZE
```

Comment:
So this value is apparently 1024 times too small. 

- #74270

I really wonder how this commit was tested.